data_structures/src/chain.rs

```diff
@@ -2568,6 +2572,28 @@ impl AltKeys {
             }
         }
     }
+
+    pub fn get_rep_ordered_bn256_list(
+        &self,
+        reputation_set: &TotalReputationSet<PublicKeyHash, Reputation, Alpha>,
+    ) -> Vec<Bn256PublicKey> {
+        let mut pkhs: Vec<PublicKeyHash> = self.bn256.iter().map(|(pkh, _)| pkh.clone()).collect();
+        pkhs.sort();
+        let ars_rep: Vec<Reputation> = pkhs.iter().map(|pkh| reputation_set.get(pkh)).collect();
+
+        let ordered_pkhs: Vec<_> = pkhs
+            .into_iter()
+            .enumerate()
+            .sorted_by_key(|(i, _x)| ars_rep[*i])
```

Comment:
`ars_rep` is not needed, you can get the reputation directly here

```suggestion
            .sorted_by_key(|pkh| reputation_set.get(pkh))
```

Reply:
Similarly you also don't need `ordered_alts` because you can add `filter_map` right after the `sorted_by_key`.

Reply:
RUST101